File: vitrivr-engine-module-m3d/src/main/java/org/vitrivr/engine/model3d/data/util/texturemodel/entropyoptimizer/EntropyOptimizerStrategy.java
package org.vitrivr.engine.model3d.data.util.texturemodel.entropyoptimizer;

/**
 * The method used to calculate the entropy.
 */
public enum EntropyOptimizerStrategy {
  /**
   * The new view vector is chosen randomly.
   */
  RANDOMIZED,
  /**
   * The new view vector is chosen by the gradient of the entropy.
   */
  NEIGHBORHOOD,
}

File: vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/mesh/texturemodel/util/entropyoptimizer/EntropyOptimizerStrategy.kt
package org.vitrivr.engine.core.model.mesh.texturemodel.util.entropyoptimizer

/**
 * The method used to calculate the entropy.
 */
enum class EntropyOptimizerStrategy {
    /**
     * The new view vector is chosen randomly.
     */
    RANDOMIZED,

    /**
     * The new view vector is chosen by the gradient of the entropy.
     */
    NEIGHBORHOOD,
}
